### ISSUE TITLE
KeyError when selecting from date IndexHierarchy with the result of iloc on the same IndexHierarchy

### ISSUE THREAD

**Issue by ForeverWintr** [InvestmentSystems/static-frame#459]
**Example**

I think this is best illustrated with an example:

```python
import static_frame as sf
import frame_fixtures as ff

f = ff.parse('f(Fg)|v(int,bool,str)|i((IY,ID),(dtY,dtD))|c(ISg,dts)|s(6,2)')

f.loc[sf.HLoc[f.index.iloc[0]]] # builtins.KeyError: 'no matching keys across all levels'
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
>>> import static_frame as sf
>>> sf.Platform.display()
<Series: platform>
<Index>
platform           Linux-5.13.0-39-generic-x86_64-with
sys.version        3.8.12 (default, Mar 31 2022, 17:04:30) [GCC 7.5.0]
static-frame       0.8.38
numpy              1.17.4
pandas             0.24.2
xlsxwriter         1.1.5
openpyxl           3.0.9
xarray             0.18.0
tables             3.6.1
pyarrow            0.17.0
msgpack            (1, 0, 0)
msgpack_numpy      None
<<U13>             <object>
```

**Comment by chaburkland** [comment]
I'll offer a more minimal example!

```python
ih = sf.IndexHierarchy.from_labels(
    [("1990", "2000-03-14")],
    index_constructors=[sf.IndexYear, sf.IndexDate]
)
ih.loc[ih.iloc[0]] # this raises a KeyError!
```

**Comment by flexatone** [comment]
Thanks for isolating this issue.

This was due to a previously identified but not handled coercion of columnar types to a single row array, forcing datetime64[Y] to datetime64[D], which when used for selection is incorrect.

The fix will be part of the 0.9.2 release.

```python
>>> ih = sf.IndexHierarchy.from_labels( 
...     [("1990", "2000-03-14")], 
...     index_constructors=[sf.IndexYear, sf.IndexDate] 
... )                                                                                                                                                                                                                                                                                                                
>>> ih.loc[ih.iloc[0]]                                                                                                                                                                                                                                                                                               
(numpy.datetime64('1990'), numpy.datetime64('2000-03-14'))
```
